services/meta/data.go

```diff
@@ -1248,12 +1248,12 @@ func (sgi *ShardGroupInfo) marshal() *internal.ShardGroupInfo {
 // unmarshal deserializes from a protobuf representation.
 func (sgi *ShardGroupInfo) unmarshal(pb *internal.ShardGroupInfo) {
 	sgi.ID = pb.GetID()
-	sgi.StartTime = UnmarshalTime(pb.GetStartTime())
-	sgi.EndTime = UnmarshalTime(pb.GetEndTime())
-	sgi.DeletedAt = UnmarshalTime(pb.GetDeletedAt())
+	sgi.StartTime = UnmarshalTime(pb.GetStartTime(), true)
```

Comment:
Rather than modifying `UnmarshalTime`, we could just add some code below that runs: `if sgi.(Start|End)Time.IsZero() { sgi.(Start|End)Time = epochTime }`